### DISCUSSION TITLE
Cannot select the GPT 4 model whereas specified in the docs

### DISCUSSION THREAD

**Discussion by sneko** [narenmanoharan/gpt-code-assistant#6] [General]
Hi,

Having a doubt but according to the docs https://github.com/narenmanoharan/gpt-code-assistant#problem it allows leveraging GPT 4 despite having the model by default to `gpt-3.5-turbo-16k`.

When using `gpt-code-assistant select-model` I only see v3.5 models. If I do `gpt-code-assistant select-model gpt-4` I get the error the model is not valid.

Can you clarify this @narenmanoharan ?

Thank you,

**Comment by narenmanoharan** [comment]
I use your API key to l[ist all the models available](https://github.com/narenmanoharan/gpt-code-assistant/blob/main/ai/open_ai.py#L44) to your from OpenAI to display the list of models you can select. Here's the [API](https://platform.openai.com/docs/api-reference/models) I use.

You can verify if you have the model available with your api key with this: 

```
curl https://api.openai.com/v1/models -H "Authorization: Bearer $OPENAI_API_KEY" | jq -r '.data[].id' | grep 'gpt'
```

Let me know if you see gpt-4 there, if so, could be a bug on the open ai python package.

**Reply by sneko** [reply]
I just get:
```
gpt-3.5-turbo-16k-0613
gpt-3.5-turbo
gpt-3.5-turbo-0301
gpt-3.5-turbo-16k
gpt-3.5-turbo-0613
```

Whereas I have OpenAI API with the paid "subscription" (and on the other hand, ChatGPT+ but it should have no link).

**Reply by sneko** [reply]
It seems all developers don't have access right away (cf. https://community.openai.com/t/api-says-gpt-4-model-doesnt-exist/289570/6). Will wait a bit more...

**Reply by sneko** [reply]
@narenmanoharan got access to gpt-4 through API (listed with your command but when doing):
`gpt-code-assistant select-model gpt-4`

I get the error: `Got unexpected extra argument (gpt-4)`

Any idea if they change something that could break your code?
